src/Adapter/Product/Combination/Repository/CombinationRepository.php

```diff
@@ -335,20 +335,63 @@ private function getAttributesInformation(array $attributeIds, int $langId): arr
 
     /**
      * @param ProductId $productId
-     * @param array $filters
+     * @param array<string, mixed> $filters
      *
      * @return QueryBuilder
      */
     private function getCombinationsQueryBuilder(ProductId $productId, array $filters): QueryBuilder
     {
-        //@todo: filters are not handled.
         $qb = $this->connection->createQueryBuilder();
         $qb->from($this->dbPrefix . 'product_attribute', 'pa')
             ->where('pa.id_product = :productId')
-            ->orderBy('id_product_attribute', 'asc')
             ->setParameter('productId', $productId->getValue())
         ;
 
+        // filter by attributes
+        if (isset($filters['attribute_ids'])) {
+            $combinationIds = $this->getCombinationIdsByAttributeIds($productId, (array) $filters['attribute_ids']);
+            $qb->andWhere($qb->expr()->in('pa.id_product_attribute', ':combinationIds'))
+                ->setParameter('combinationIds', $combinationIds, Connection::PARAM_INT_ARRAY)
+            ;
+        }
+
+        $qb->orderBy('id_product_attribute', 'asc');
+
         return $qb;
     }
+
+    /**
+     * @param ProductId $productId
+     * @param int[] $attributeIds
+     *
+     * @return int[]
+     */
+    private function getCombinationIdsByAttributeIds(ProductId $productId, array $attributeIds): array
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb->select('pac.id_product_attribute')
+            ->from($this->dbPrefix . 'product_attribute_combination', 'pac')
+            ->leftJoin(
+                'pac',
+                $this->dbPrefix . 'product_attribute',
+                'pa',
+                'pac.id_product_attribute = pa.id_product_attribute'
+            )
+            ->where('pa.id_product = :productId')
+            ->setParameter('productId', $productId->getValue())
+            ->andWhere($qb->expr()->in('pac.id_attribute', ':attributeIds'))
```

Comment:
Oh interesting how you set the parameter for `IN` clause I had to implement it recently in one of my PR but I used `$qb->expr()->in('pac.id_attribute', $attributeIds)` directly

I like your way better, I'll do the same, thanks!